src/battle.ts

```diff
@@ -3539,10 +3546,15 @@ export class Battle {
 		}
 		case 'poke': {
 			let pokemon = this.rememberTeamPreviewPokemon(args[1], args[2])!;
-			if (args[3] === 'mail') {
-				pokemon.item = '(mail)';
-			} else if (args[3] === 'item') {
-				pokemon.item = '(exists)';
+			if (args[3]) {
+				const notes = JSON.parse(args[3]);
+				Object.assign(pokemon, notes);
```

Comment:
Merging a malformed object received from the server may break the ``pokemon`` instance.

Reply:
Specifically, before I forget, ``pokemon.item`` must be escaped.

Reply:
I will restore the existing code for the mods and add checks for Tera Type Preview and Force Open Team Sheets.

Assuming that a `details`-esque string is sent instead of JSON, should the name be sent as an additional `poke` parameter in case the `item` contains something that invalidates splitting `name` and `item` at `lastIndexOf(';item:')` ?

Are there other formatting concerns in this context, such as other properties that might include `;`?

Reply:
``;`` doesn't have a reserved meaning for any of PS protocols, IIRC, so it would be best not to use it. Instead, ``,`` or ``|`` can be used. Or even hyphen ``-`` or spaces, if the context is non-ambiguous.

Normally, we'd just add a new ``args`` parameter with ``|``, but you may prefer using ``kwArgs`` (i.e. ``|tera: Poison``), specially since it seems that ``|poke|`` may get an increasingly amount of extra specifiers in further generations.

Reply:
(You may also use JSON, too, but the amount of safeguards that need to be added make it not a very pretty approach.)

Reply:
Sorry, @qwertystrargs , it seems like I forgot how kwargs work. Syntax is ``|[tera] Poison``. Then you may access ``kwArgs.tera == 'Poison'``, and ``args[3]`` can still be used for items exclusively.